### PR TITLE
moving the use of modified GET discord-actions/groups api under feature flag

### PR DESCRIPTION
**Pull request by bksh05**
### Description
This pull request aims to address the issue of discrepancies between the `GET discord-actions/groups` API in STAGING and PROD environments. Currently, the API is functioning correctly on STAGING, but it returns an Internal Server Error on PROD. While the exact root cause is still under investigation, there is a consensus that the issue might be related to stale data in the PROD environment.
Fixes: #450 

#### Proposed Solution
To resolve this issue and ensure a controlled testing and debugging process, a feature flag mechanism is being introduced to the GET discord-actions/groups API. This feature flag will allow the gradual release of new changes to the API on different environments while maintaining stability in the existing functionality.

### Changes
The changes utilize the `dev=true` flag from the URL and subsequently pass this flag to the API. Depending on the value of this flag, the API will generate a list of groups, adjusting whether additional membership data is included in the response or not.

#### Before
image

##### Without flag 
image

##### With flag (dev=true)
image

### Demo Video

https://github.com/Real-Dev-Squad/website-dashboard/assets/27444637/a509a093-40fd-44d5-97cb-d4dd7c4b484e

### CODE REVIEW

**Review by iamitprakash — DISMISSED** [review]

File: groups/script.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ const sections = document.querySelectorAll('.manage-groups, .create-group');
 const loader = document.querySelector('.backdrop');
 const userIsNotVerifiedText = document.querySelector('.not-verified-tag');
 const userSelfData = await getUserSelf();
+const params = new URLSearchParams(window.location.search);
+const isDev = params.get('dev');
```
Comment (RitikJaiswal75) [inline groups/script.js:17]: lets match its value to be true 
else tomorrow I can just write dev='someRandomStuff" and is dev will still be considered

**Review by RitikJaiswal75 — COMMENTED** [review]
Please add tests for the feature

**Comment by bksh05** [comment]
> Please add tests for the feature

There are tests for the feature already. 
I am only moving this under the feature flag as part of this PR

File: groups/script.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ const sections = document.querySelectorAll('.manage-groups, .create-group');
 const loader = document.querySelector('.backdrop');
 const userIsNotVerifiedText = document.querySelector('.not-verified-tag');
 const userSelfData = await getUserSelf();
+const params = new URLSearchParams(window.location.search);
+const isDev = params.get('dev') === 'true';
```
Comment (vvaibhavdesai) [inline groups/script.js:17]: move these strings into a constant as they are easier to maintain then

File: groups/script.js [hunk 3]
```diff
@@ -116,7 +118,7 @@ groupRoles?.addEventListener('click', function (event) {
   if (groupListItem) {
     const newURL = `${window.location.pathname}?${
       groupListItem.querySelector('p').textContent
-    }`;
+    }${isDev ? '&dev=true' : ''}`;
```
Comment (vvaibhavdesai) [inline groups/script.js:121]: `const isDevModeEnabled = isDev ? '&dev=true' : ''`

File: groups/utils.js [hunk 4]
```diff
@@ -32,15 +32,18 @@ async function getUserSelf() {
...
-async function getDiscordGroups() {
+async function getDiscordGroups(isDev) {
   try {
-    const res = await fetch(`${BASE_URL}/discord-actions/groups`, {
-      method: 'GET',
-      credentials: 'include',
-      headers: {
-        'Content-type': 'application/json',
+    const res = await fetch(
+      `${BASE_URL}/discord-actions/groups${isDev ? '?dev=true' : ''}`,
```
Comment (vvaibhavdesai) [inline groups/utils.js:38]: same as above

**Review by prakashchoudhary07 — DISMISSED** [review]

File: groups/script.js (hunk 3 above)
Comment (bksh05) [reply groups/script.js]: fixed

File: groups/script.js (hunk 2 above)
Comment (bksh05) [reply groups/script.js]: done

File: groups/utils.js (hunk 4 above)
Comment (bksh05) [reply groups/utils.js]: done

**Comment by bksh05** [comment]
> Can you please show the working video of the feature with all the network requests and also add the test coverage to your PR?

Hi @Pratiyushkumar ,
I have added the video you asked for. 
The test coverage is not generated for the dashboard application. So, I haven't attached it. 

Also, I am just moving the changes under the feature flag and the feature-related test cases are already written as part of the previous PR.

**Review by prakashchoudhary07 — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]
LGTM

**Pull request merged**